### DISCUSSION TITLE
How to troubleshoot a maybe-language-server issue?

### DISCUSSION THREAD

**Discussion by MatrixManAtYrService** [helix-editor/helix#9684] [Troubleshooting] [answered]
Here's an example of a bug which I suspect could be a language server problem.  The deletion of the highlighted text makes the problem go away.

**Before** (note that `sleep` is white):

<img width="545" alt="Screenshot 2024-02-20 at 6 53 19 PM" src="https://github.com/helix-editor/helix/assets/5834582/122a47e2-dd57-4def-b4f7-cbf918d7b954">

**After** (note that `sleep` is green):

<img width="584" alt="Screenshot 2024-02-20 at 6 53 27 PM" src="https://github.com/helix-editor/helix/assets/5834582/5c1f882f-2f0a-4b30-b337-29f124173da7">

Some character in that box drawing thingy that I generated from https://asciiflow.com/#/ appears to be breaking syntax highlighting: 

```
 ┌──────┐             GET                 ┌──────────────┐
 │ Core ├────────────────────────────┐    │  Laminar     │
 └───▲──┘                            │    │  ┌───────────┤
     │                               └────┼──► Routes    │
 GET │    request from Core               │  │ ┌──────┐  │
     │    which will request from Laminar │  │ │ Mocks│  │
```

----
**Edit:**  After a bit more scrutiny, I see that the same weird behavior appears when github parses the code:

```python3
from time import sleep

def main() -> None:
    foo(
        json={"code": 200, "response": data.dict(), #  <-- missing '}'
        headers={"Content-Type": "application/json"}
    )
    sleep(1) #<-- purple
    sleep(1)
    sleep(1) #<-- white
    sleep(1)
```

So it probably is a language server issue and not a helix issue.  But generally speaking, how to know the difference?

**Comment by kirawi** [comment]
This looks like https://github.com/helix-editor/helix/issues/1151

**Reply by MatrixManAtYrService** [reply]
It does, but while trying to come up with a helpful comment in that issue, I stumbled across evidence that it's probably not a helix bug, see my edit ☝️

**Comment by the-mikedavis** [comment, marked as answer]
Syntax highlighting in Helix (and GitHub for some languages including Python) is done by tree-sitter rather than LSP. I expect that you should be able to reproduce that behavior with the same code running in a scratch buffer (`:new`, `:lang python`, then paste in the contents) since scratch buffers don't use LSP. To debug it would be useful to minimize the behavior down to the smallest file that can reproduce the behavior
